mlflow/entities/lifecycle_stage.py

```diff
@@ -0,0 +1,35 @@
+from mlflow.entities.view_type import ViewType
+from mlflow.exceptions import ExecutionException
+
+
+class LifecycleStage(object):
+    ACTIVE = "active"
+    DELETED = "deleted"
+    _VALID_STAGES = set([ACTIVE, DELETED])
+
+    @classmethod
+    def view_type_to_stages(cls, view_type=ViewType.ALL):
+        stages = []
+        if view_type == ViewType.ACTIVE_ONLY or view_type == ViewType.ALL:
+            stages.append(cls.ACTIVE)
+        if view_type == ViewType.DELETED_ONLY or view_type == ViewType.ALL:
+            stages.append(cls.DELETED)
+        return stages
+
+    @classmethod
+    def is_valid(cls, lifecycle_stage):
+        return lifecycle_stage in cls._VALID_STAGES
+
+    @classmethod
+    def matches_view_type(cls, view_type, lifecycle_stage):
+        if not cls.is_valid(lifecycle_stage):
+            raise ExecutionException("Invalid lifecycle stage '%s'" % str(lifecycle_stage))
```

Comment:
The documentation for `ExecutionException` doesn't seem to fit this use case:

```
class ExecutionException(MlflowException):
    """Exception thrown when executing a project fails."""
    pass
```

Since the exception case seems most appropriate for an "internal" exception (one that doesn't inherit from `MlflowException`), I think that `ValueError` may be a better fit.

Reply:
Right! Moved this code from `RunInfo`.
I will just go with an `MlflowException`